### PR TITLE
docs: add mfa platform docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* Describes the process for enabling / disabling MFA for a user account on Supabase

### CODE REVIEW

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  title: 'Multi-factor Authentication',
+  description: 'Enable multi-factor authentication (MFA) to keep your account secure.',
+  subtitle: 'Enable multi-factor authentication (MFA) to keep your account secure.',
+}
+
+Multi-factor authentication (MFA) adds an additional layer of security to your organization and prevents unauthorized access to your user account. MFA improves the security of your account by requiring a second factor to verify your user identity. Supabase allows users to enable MFA on their account and set it as a requirement for subsequent logins.
+
+## Supported Authentication Factors
+
+Currently, Supabase supports adding a unique time-based one-time password (TOTP) to your user account as an additional security factor. You can manage your TOTP factor using apps such as 1Password, Authy, Google Authenticatior or Apple's Keychain.
```
Comment (J0) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:14]:
```suggestion
Currently, Supabase supports adding a unique time-based one-time password (TOTP) to your user account as an additional security factor. You can manage your TOTP factor using apps such as 1Password, Authy, Google Authenticator or Apple's Keychain.
```

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Currently, Supabase supports adding a unique time-based one-time password (TOTP) to your user account as an additional security factor. You can manage your TOTP factor using apps such as 1Password, Authy, Google Authenticator or Apple's Keychain.
+
+## Enable MFA
+
+You can enable MFA for your user account under your [Supabase account settings](/dashboard/account/security). Enabling MFA will result in all other user sessions to be automatically logged out and forced to sign-in again with MFA.
+
+<Admonition type="note">
+
+Supabase does not support returning recovery codes for MFA. Instead, we recommend that you register a backup TOTP factor to use in an event that you lose access to your primary TOTP factor.
```
Comment (J0) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:22]:
```suggestion
Supabase does not return recovery codes. Instead, we recommend that you register a backup TOTP factor to use in an event that you lose access to your primary TOTP factor.
```

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## Disable MFA
+
+You can disable MFA for your user account under your [Supabase account settings](/dashboard/account/security). On subsequent login attempts, you will not be prompted to enter a 2FA code.
+
+<Admonition type="caution">
+
+We strongly recommend that you do not disable MFA to avoid unauthorized access to your user account.
+
+</Admonition>
```
Comment (J0) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:38]: Should we add a section on what to do in the event that they lose their backup MFA device?

(e.g. do reach out to support via the support form if you lose access to both your primary and backup MFA factors . Note that for security reasons, a MFA reset may take upwards of 3 days to process)

**Review by J0 — APPROVED** [review]
LGTM, thanks for putting this together

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx (hunk 3 above)
Comment (kangmingtay) [reply apps/docs/pages/guides/platform/multi-factor-authentication.mdx:38]: yeah good point

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+import { IconCheck } from 'ui'
+
+export const meta = {
+  title: 'Multi-factor Authentication',
+  description: 'Enable multi-factor authentication (MFA) to keep your account secure.',
+  subtitle: 'Enable multi-factor authentication (MFA) to keep your account secure.',
+}
+
+Multi-factor authentication (MFA) adds an additional layer of security to your organization and prevents unauthorized access to your user account. MFA improves the security of your account by requiring a second factor to verify your user identity. Supabase allows users to enable MFA on their account and set it as a requirement for subsequent logins.
```
Comment (hf) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:10]: Given all the docs feedback for SSO, can you add an admonition here something like:

> This guide is for managing MFA on the Supabase dashboard. For adding MFA capabilities to your Supabase project go to ...

And the reverse on the MFA guide for projects too.

Comment (hf) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:10]:
```suggestion
}

Multi-factor authentication (MFA) adds an additional layer of security to your user account, by requiring a second factor to verify your user identity. Supabase allows users to enable MFA on their account and set it as a requirement for subsequent logins.
```

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+<Admonition type="note">
+
+Supabase does not return recovery codes. Instead, we recommend that you register a backup TOTP factor to use in an event that you lose access to your primary TOTP factor.
+
+</Admonition>
+
+## Login with MFA
+
+Once you've enabled MFA for your Supabase user account, you will be prompted to enter your two-factor (2FA) authentication code. You can find your unique code generated by your preferred TOTP app.
```
Comment (hf) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:28]: Is it 2FA or MFA?

```suggestion
Once you've enabled MFA for your Supabase user account, you will be prompted to enter your second factor challenge code as seen in your preferred TOTP app.
```

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Disable MFA
+
+You can disable MFA for your user account under your [Supabase account settings](/dashboard/account/security). On subsequent login attempts, you will not be prompted to enter a 2FA code.
```
Comment (hf) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:32]:
```suggestion
You can disable MFA for your user account under your [Supabase account settings](/dashboard/account/security). On subsequent login attempts, you will not be prompted to enter a MFA code.
```

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+Currently, Supabase supports adding a unique time-based one-time password (TOTP) to your user account as an additional security factor. You can manage your TOTP factor using apps such as 1Password, Authy, Google Authenticator or Apple's Keychain.
+
+## Enable MFA
+
+You can enable MFA for your user account under your [Supabase account settings](/dashboard/account/security). Enabling MFA will result in all other user sessions to be automatically logged out and forced to sign-in again with MFA.
+
+<Admonition type="note">
+
+Supabase does not return recovery codes. Instead, we recommend that you register a backup TOTP factor to use in an event that you lose access to your primary TOTP factor.
```
Comment (hf) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:22]:
```suggestion
Supabase does not return recovery codes. Instead, we recommend that you register a backup TOTP factor to use in an event that you lose access to your primary TOTP factor. Make sure you use a different device and app, or store the secret in a secure location different than your primary one.
```

File: apps/docs/pages/guides/platform/multi-factor-authentication.mdx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  title: 'Multi-factor Authentication',
+  description: 'Enable multi-factor authentication (MFA) to keep your account secure.',
+  subtitle: 'Enable multi-factor authentication (MFA) to keep your account secure.',
+}
+
+Multi-factor authentication (MFA) adds an additional layer of security to your organization and prevents unauthorized access to your user account. MFA improves the security of your account by requiring a second factor to verify your user identity. Supabase allows users to enable MFA on their account and set it as a requirement for subsequent logins.
+
+## Supported Authentication Factors
+
+Currently, Supabase supports adding a unique time-based one-time password (TOTP) to your user account as an additional security factor. You can manage your TOTP factor using apps such as 1Password, Authy, Google Authenticator or Apple's Keychain.
```
Comment (hf) [inline apps/docs/pages/guides/platform/multi-factor-authentication.mdx:14]:
```suggestion
Currently, Supabase supports adding a unique time-based one-time password (TOTP) to your user account as an additional security factor. You can manage your TOTP factor using apps such as 1Password, Authy, Google Authenticator or Apple's Keychain.
```

**Review by hf — APPROVED** [review]

**Pull request merged**